src/scales/scale.time.js

```diff
@@ -404,25 +402,27 @@ function ticksFromTimestamps(values, majorUnit) {
 	return ticks;
 }
 
-function determineLabelFormat(data, timeOpts) {
-	var i, momentDate, hasTime;
-	var ilen = data.length;
+/**
+ * Return the time format for the label with the most parts (milliseconds, second, etc.)
+ */
+function determineLabelFormat(timestamps) {
+	var presets = adapter.presets(true);
```

Comment:
Could we have `adapter.LONG_PRESETS` here instead? Currently, it's not clear what the `true` argument is without opening up the adapter interface, so that might be slightly more readable

Reply:
I agree it's not as clear as your suggestion but actually, I'm not happy with the short vs long concept. It makes the abstraction more complicated and I would prefer to get rid of it and keep `presets()` without arguments.

These long presets have been introduced by #5095 to implement *a specific use case*, hard coded and that can only be changed globally with `tooltipFormat`. [I've never been convinced by this approach](/chartjs/Chart.js/pull/5095#issuecomment-356279455) and @kurkle made me notice that most of the time, users will override `tooltipFormat`.

So instead to keep making things more complex, we should revert #5095 and display the same format as the tick label ([suggested by the original issue reporter](1466)). This would simplify the adapter API, but also the time scale logic. Implementing your short / long preset logic can easily be done in the financial chart using a plugin or overriding specific methods. If at some point we introduce scriptable option for the tooltip format, it will even be easier.

Reply:
Whether to simplify the tooltip and whether to have `SHORT_PRESETS` and `LONG_PRESETS` might be different issues. The reason I say that is that it wouldn't be possible to implement this logic externally in a method that is agnostic to the time-library used unless we still have `SHORT_PRESETS` and `LONG_PRESETS`. Otherwise, anyone who wanted to implement such logic would need to create a `LONG_PRESETS` for every time library and include it in their own plugin. So I think it is a nice feature to include both

Reply:
This adapter is private (we don't want backward compatibility) so "external code" should not rely on this adapter to implement new features, at least not until we make it public. 

Simplifying the tooltip will make this adapter simpler and lighter by removing superfluous API (and it's not only these presets maps, see @kurkle comment). Any evolution of this adapter to support new features should be discussed separately and implemented later, let's stay focus on the initial goal which is to get Chart.js working with Luxon and others.

Reply:
Anyway, since this adapter is private, let's add the extra API to keep supporting #5095 and remove later if we simplify the tooltip behavior.

Reply:
Ok. Makes sense to me to get this in and then handle the other issues separately

Reply:
I took another direction to make this call clearer, it's now `formats()` to fetch tokens per unit and `presets()` to fetch the 3 versions introduced by #5095.